frontend/src/pages/notebookController/StartServerModal.tsx

```diff
@@ -1,28 +1,54 @@
 import * as React from 'react';
-import {
-  ActionList,
-  ActionListItem,
-  Button,
-  Modal,
-  ModalVariant,
-  Spinner,
-} from '@patternfly/react-core';
-import NotebookControllerContext from './NotebookControllerContext';
+import { Alert, AlertVariant, Button, Modal, ModalVariant, Spinner } from '@patternfly/react-core';
+import { Notebook } from '../../types';
+import { checkNotebookRunning } from '../../utilities/notebookControllerUtils';
 
 import './NotebookController.scss';
 
 type StartServerModalProps = {
+  notebook: Notebook | undefined;
   startShown: boolean;
   setStartModalShown: (shown: boolean) => void;
   onClose: () => void;
 };
 
+type SpawnStatus = {
+  status: AlertVariant;
+  title: string;
+  reason: React.ReactNode;
+};
+
 const StartServerModal: React.FC<StartServerModalProps> = ({
+  notebook,
   startShown,
   setStartModalShown,
   onClose,
 }) => {
-  const { notebook, isNotebookRunning } = React.useContext(NotebookControllerContext);
+  const [spawnStatus, setSpawnStatus] = React.useState<SpawnStatus | null>(null);
+
+  const isNotebookRunning = checkNotebookRunning(notebook);
+
+  React.useEffect(() => {
+    let timer;
+    if (isNotebookRunning) {
+      setSpawnStatus({
+        status: AlertVariant.success,
+        title: 'Success',
+        reason: 'The notebook server is up and running. This page will update momentarily.',
+      });
+      timer = setTimeout(() => {
+        if (notebook?.metadata.annotations?.['opendatahub.io/link']) {
+          window.location.href = notebook.metadata.annotations['opendatahub.io/link'];
+        }
```

Comment:
I thought we agreed on some notification or something when we don't fallback on something. Perhaps we convert the modal to an error message?
```
else {
  setSpawnStatus({
    status: AlertVariant.danger,
    title: 'Failed to redirect',
    reason: 'For unknown reasons the notebook server was unable to be redirected to.',
  });
}
```

If you want to notification scenario... close the modal & kick the notification. They are on the Control Panel, they can enter the notebook manually (hopefully?).

Either way, you should always handle your error states.

Reply:
Updated.